backend/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var clientset *kubernetes.Clientset
```

Comment:
I know it's heavily overkill at this stage, but in the production version it will be hard for us to use this code as is. clientSet will probably be created in a different file than the endpoints will be, making it hard to ensure that it is always set to connect to the cluster the application is running on before the endpoint is called, and you can't leave it so that the connection is always occur in the main function, because that won't work for testing. 
Such a flagship solution would be to use the singleton design pattern here, but according to chatgpt golang has something like sync.Once and it should also work here. Then you could create a function to retrieve the connected clientset that would be accessible from anywhere, and make sure we only have one connection open.

Reply:
Created wrapped in singleton in different file